[Update] Remove all Classic Manager guides from the library

leslitagordita opened the pull request:
* removes all guides with `-classic-manager` appended
* Adds aliases from classic to cloud guide
* Removes `classic_manager_link` front matter from any cloud guides

nmelehan commented:
Deploy preview for *nostalgic-ptolemy-b01ab8* ready!

Built with commit 3be77a0ce49463420a95ec5fbed65c185d7f0102

https://deploy-preview-3100--nostalgic-ptolemy-b01ab8.netlify.com

leslitagordita commented:
To add all `-classic-manager` aliases to cloud guides I used functions from Nate's `mv-guide.py` script. This script reads a list of directory locations from a txt file.

```
#! /usr/local/bin/python3

import sh
import sys
import os

script_location = os.path.dirname(os.path.abspath(__file__))
repo_dir = script_location
content_dir = os.path.join(repo_dir, 'docs')

def parse_aliases(alias_line):
    aliases_as_messy_strings = alias_line.split(':')[-1].strip().split(',')

    aliases_as_clean_strings = map(
        lambda x: x.strip(" '[]"), aliases_as_messy_strings)

    return set(map(lambda x: os.path.normpath(x).lower() + "/", aliases_as_clean_strings))

def alias_style_path_for_path(content_dir, path):
    directory = os.path.dirname(path) if os.path.isfile(path) else path
    directory_abs_path = os.path.abspath(directory)
    return os.path.relpath(directory_abs_path, content_dir).lower() + "/"

def insert_aliases(content_dir, paths):
    for path in paths:
        basename = os.path.basename(path)
        if os.path.isfile(path) and (basename == 'index.md' or basename == '_index.md'):
            insert_alias(content_dir, path)
        elif os.path.isdir(path):
            insert_aliases(content_dir, path_contents(path))

def insert_alias(content_dir, markdown_file):
    def serialized_alias_set(aliasSet):
        return "[" + ",".join(map(lambda x: "'" + x + "'", aliasSet)) + "]"

    new_alias = markdown_file
    new_alias_append = new_alias + '-classic-manager/'
    markdown_file_append = 'docs/' + markdown_file + '/index.md'

    with open(markdown_file_append, 'r+') as fd:
        contents = fd.readlines()
        fd.seek(0)
        frontmatter_begin = False
        frontmatter_line_written = False
        for line in contents:
            if "aliases: " in line and frontmatter_begin:
                aliases = parse_aliases(line) | {new_alias_append}
                fd.write("aliases: " + serialized_alias_set(aliases) + "\n")
                frontmatter_line_written = True
            elif "---" in line and frontmatter_begin and not frontmatter_line_written:
                fd.write("aliases: " +
                         serialized_alias_set({new_alias_append}) + "\n" + line)
                frontmatter_line_written = True
            elif "---" in line:
                frontmatter_begin = True
                fd.write(line)
            else:
                fd.write(line)
        fd.truncate()


file = open('directories.txt', 'r')
locations = file.readlines()

for i in locations:
    insert_alias(content_dir, i.rstrip('\n'))

```
The text file included these directories:
```
quick-answers/linode-platform/add-caa-dns-records
quick-answers/linode-platform/find-your-linodes-ip-address
quick-answers/linode-platform/reset-the-root-password-on-your-linode
quick-answers/linode-platform/resize-a-linode-disk
quick-answers/linode-platform/deploy-an-image-to-a-linode
networking/dns/previewing-websites-without-dns
networking/dns/configure-your-linode-for-reverse-dns
networking/dns/common-dns-configurations
platform/block-storage/how-to-use-block-storage-with-your-linode
platform/linode-managed
platform/stackscripts
platform/manager/remote-access
platform/manager/using-the-linode-shell-lish
platform/manager/accounts-and-passwords
platform/manager/dns-manager
platform/manager/using-the-linode-graphical-shell-glish
platform/api/getting-started-with-the-linode-api
platform/nodebalancer/nodebalancer-reference-guide
platform/nodebalancer/getting-started-with-nodebalancers
platform/disk-images/linode-images
platform/disk-images/linode-backup-service
platform/disk-images/disk-images-and-configuration-profiles
platform/disk-images/resizing-a-linode
platform/disk-images/clone-your-linode
platform/billing-and-support/support
platform/billing-and-support/tax-information
platform/billing-and-support/billing-and-payments
platform/network-helper
troubleshooting/rescue-and-rebuild
getting-started
uptime/monitoring-and-maintaining-your-server
```

andystevensname commented on docs/platform/longview/what-is-longview/index.md:
```diff
@@ -5,15 +5,14 @@ author:
 keywords: ["system monitoring", "longview", "metrics"]
 license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
-aliases: ['platform/longview/longview/']
+aliases: ['platform/longview/longview/','longview/', 'uptime/longview/']
```
Should this include the classic manager alias as well?

leslitagordita replied:
No, because the `longview-classic-manager` never existed as a guide. Classic manager version before the cloud manager version was created was: `platform/longview/longview/`.

leslitagordita commented on docs/networking/dns/common-dns-configurations/index.md:
```diff
@@ -6,13 +6,12 @@ description: 'Configurations for common DNS records.'
 keywords: ["dns", "dnssec"]
 license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
-aliases: ['dns-manager/','dns-guides/configuring-dns-with-the-linode-manager/']
+aliases: ['dns-guides/configuring-dns-with-the-linode-manager/','networking/dns/common-dns-configurations-classic-manager/','dns-manager/']
```
I found that serveral DNS guides were aliasing `dns-guides/configuring-dns-with-the-linode-manager/`. I removed the alias from two guides and kept it in the Common DNS Configurations guide....fixed

leslitagordita commented on docs/networking/dns/configure-your-linode-for-reverse-dns/index.md:
```diff
@@ -5,13 +5,12 @@ author:
 keywords: ["reverse", "dns", "PTR"]
 license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
-aliases: ['networking/Setting-Up-Reverse-DNS-Lookup/','dns-guides/configuring-dns-with-the-linode-manager/','networking/configure-your-linode-for-reverse-dns/','networking/dns/setting-reverse-dns/','dns-manager/']
+aliases: ['networking/configure-your-linode-for-reverse-dns/','dns-manager/','networking/dns/setting-reverse-dns/','networking/setting-up-reverse-dns-lookup/','networking/dns/configure-your-linode-for-reverse-dns-classic-manager/','dns-guides/configuring-dns-with-the-linode-manager/']
```
Those aliases were not added as a result of this PR, BUT should be updated with the changes you suggest...fixed.

leslitagordita commented:
@andystevensname pushed up a fix for the 404, which does the following things:
* Added missing alias to guide.
* Updated 404ing link to link to Events and Activity feed guide 
* Added a new section to Events and Activity feed guide for managing events email notifications

leslitagordita commented:
see new PR: https://github.com/linode/docs/pull/3113
